lxd/db/cluster/certificates.go

```diff
@@ -115,6 +115,22 @@ func (cert Certificate) ToIdentity() (*Identity, error) {
 	return identity, nil
 }
 
+var getCertificateIdentitiesStmt = fmt.Sprintf(`
+SELECT identities.id, identities.auth_method, identities.type, identities.identifier, identities.name, identities.metadata
+	FROM identities
+	WHERE auth_method = %d
+	AND type in (%d, %d, %d, %d, %d)
+`, authMethodTLS,
+	identityTypeCertificateServer,
+	identityTypeCertificateClientRestricted,
+	identityTypeCertificateClientUnrestricted,
+	identityTypeCertificateMetricsRestricted,
+	identityTypeCertificateMetricsUnrestricted,
+)
+
+var getCertificateIdentityByFingerprintStmt = fmt.Sprintf(`%s AND identifier = ?`, getCertificateIdentitiesStmt)
```

Comment:
Rather than store the query 3 times in memory, can you generate the query inside the relevant function using just getCertificateIdentitiesStmt as the basis?

Also, there's no need to use `fmt.Sprintf` as that uses reflection to detect the input type, instead you can just use `+` to concatenate the strings.

Reply:
Yep will do.

Reply:
Done.